mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -79,9 +79,6 @@ abstract class StreamingLinearAlgorithm[
    * @param data DStream containing labeled data
    */
   def trainOn(data: DStream[LabeledPoint]): Unit = {
-    if (model.isEmpty) {
```

Comment:
No, I think the user must set the weights before starting. Above it says "Initial weights must be set before using trainOn or predictOn". I am not sure it makes sense for the scaladoc to talk about a default, therefore. Yes `Vectors.dense(numFeatures)` is a bug but I suppose it is never hit since the model is always defined at this point. I think this can be changed along with the comment. Yeah, that is probably worth touching up.